controllers/spacecompletion/space_completion_controller_test.go

```diff
@@ -0,0 +1,203 @@
+package spacecompletion_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/codeready-toolchain/host-operator/controllers/spacecompletion"
+	"github.com/codeready-toolchain/host-operator/pkg/apis"
+	"github.com/codeready-toolchain/host-operator/pkg/counter"
+	. "github.com/codeready-toolchain/host-operator/test"
+	spacetest "github.com/codeready-toolchain/host-operator/test/space"
+	"github.com/codeready-toolchain/toolchain-common/pkg/cluster"
+	"github.com/codeready-toolchain/toolchain-common/pkg/configuration"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func TestCreateSpace(t *testing.T) {
+	member1 := NewMemberCluster(t, "member1", corev1.ConditionTrue)
+	getMemberClusters := NewGetMemberClusters(member1)
+
+	t.Run("without any field set - then it sets only tierName", func(t *testing.T) {
+		// given
+		space := spacetest.NewSpace("without-fields",
+			spacetest.WithTierName(""))
+		r, req, cl := prepareReconcile(t, space, getMemberClusters)
+
+		// when
+		_, err := r.Reconcile(context.TODO(), req)
+
+		// then
+		require.NoError(t, err)
+		spacetest.AssertThatSpace(t, test.HostOperatorNs, space.Name, cl).
```

Comment:
could you add an assertion on the response, in particular, on the `requeue` field? (assuming we expect `false` at this point?)

Reply:
(same comment applies on all tests below)

Reply:
it's supposed to be false literally in all cases, so I followed the same approach as we use in all other controllers like that - I don't check that.